modules/opcua_server_module/src/opcua_server_impl.cpp

```diff
@@ -26,25 +26,72 @@ OpcUaServerImpl::~OpcUaServerImpl()
 {
 }
 
-PropertyObjectPtr OpcUaServerImpl::createDefaultConfig()
+void OpcUaServerImpl::populateDefaultConfigFromProvider(const ContextPtr& context, const PropertyObjectPtr& config)
+{
+    if (!context.assigned())
+        return;
+    if (!config.assigned())
+        return;
+
+    LoggerComponentPtr loggerComponent;
+
+    auto options = context.getModuleOptions("OpcUaServer");
+    for (const auto& [key, value] : options)
+    {
+        if (config.hasProperty(key))
+        {
+            ErrCode err = config->setPropertyValue(key, value);
+            if (err != OPENDAQ_SUCCESS)
+            {
+                loggerComponent = context.getLogger().getOrAddComponent("ConfigProvider/Modules/OpcUaServer");
+                LOG_W("Ignoring property \"{}\". Using default value \"{}\"", key, config.getPropertyValue(key));
+            }
+        }
+        else
+        {
+            auto property = ObjectPropertyBuilder(key, value).setValueType(value.getCoreType()).build();
+            config.addProperty(property);
+        }
+    }
+}
+
+PropertyObjectPtr OpcUaServerImpl::createDefaultConfig(const ContextPtr& context)
 {
     constexpr Int minPortValue = 0;
     constexpr Int maxPortValue = 65535;
 
     auto defaultConfig = PropertyObject();
 
+    defaultConfig.addProperty(StringProperty("Name", "OpenDAQ_Server"));
+    defaultConfig.addProperty(StringProperty("Manufacturer", "openDAQ"));
+    defaultConfig.addProperty(StringProperty("Model", ""));
+    defaultConfig.addProperty(StringProperty("SerialNumber", ""));
+    defaultConfig.addProperty(BoolProperty("ServiceDiscoverable", false));
+    defaultConfig.addProperty(StringProperty("ServicePath", "/"));
```

Comment:
These should just use the info from the root device.

Reply:
agree